src/Appwrite/Platform/Workers/Audits.php

```diff
@@ -103,27 +104,42 @@ public function action(Message $message, Database $dbForProject): void
             'timestamp' => DateTime::formatTz(DateTime::now())
         ];
 
-        $this->logs[] = $eventData;
+        if (isset($this->logs[$project->getInternalId()])) {
+            $this->logs[$project->getInternalId()]['logs'][] = $eventData;
+        } else {
+            $this->logs[$project->getInternalId()] = [
+                'project' => new Document([
+                    '$id' => $project->getId(),
+                    '$internalId' => $project->getInternalId(),
+                    'database' => $project->getAttribute('database'),
+                ]),
+                'logs' => [$eventData]
+            ];
+        }
 
         // Check if we should process the batch by checking both for the batch size and the elapsed time
         $batchSize = $this->getBatchSize();
-        $shouldProcessBatch = count($this->logs) >= $batchSize;
-        if (!$shouldProcessBatch && count($this->logs) > 0) {
-            $shouldProcessBatch = (time() - $this->lastTriggeredTime) >= self::BATCH_AGGREGATION_INTERVAL;
+        $shouldProcessBatch = \count($this->logs) >= $batchSize;
+        if (!$shouldProcessBatch && \count($this->logs) > 0) {
+            $shouldProcessBatch = (\time() - $this->lastTriggeredTime) >= self::BATCH_AGGREGATION_INTERVAL;
         }
 
         if ($shouldProcessBatch) {
-            Console::log('Processing batch with ' . count($this->logs) . ' events');
-            $audit = new Audit($dbForProject);
-
             try {
-                $audit->logBatch($this->logs);
-                Console::success('Audit logs processed successfully');
+                foreach ($this->logs as $internalId => $projectLogs) {
```

Comment:
I would rename from `internalId to `index`. This name might misslead into thinking its ID from DB, while it's just array index